### PR TITLE
Orchestrator config io

### PR DESCRIPTION
**Pull request by afanfa**
This PR includes the orchestrator layer, the configuration module and the IO layer.

### CODE REVIEW

File: trunk/NDHMS/CPL/CLM_cpl/module_clm_HYDRO.F [hunk 1]
```diff
@@ -54,7 +54,8 @@ subroutine clm_cpl_HYDRO()
        use module_rt_data, only:  rt_domain
        use module_CPL_LAND, only: CPL_LAND_INIT, cpl_outdate
        use module_mpp_land
-       use module_namelist, only: nlst_rt 
+       !use module_namelist, only: nlst_rt
```
Comment (jmccreight) [inline trunk/NDHMS/CPL/CLM_cpl/module_clm_HYDRO.F]: should remove it

File: trunk/NDHMS/CPL/LIS_cpl/module_lis_HYDRO.F [hunk 2]
```diff
@@ -30,7 +30,8 @@ module module_lis_HYDRO
        use module_rt_data, only:  rt_domain
        use module_CPL_LAND, only: CPL_LAND_INIT, cpl_outdate
        use module_mpp_land
-       use module_namelist, only: nlst_rt
+       !use module_namelist, only: nlst_rt
```
Comment (jmccreight) [inline trunk/NDHMS/CPL/LIS_cpl/module_lis_HYDRO.F]: remove

File: trunk/NDHMS/CPL/NoahMP_cpl/hrldas_drv_HYDRO.F [hunk 3]
```diff
@@ -77,7 +77,8 @@ subroutine HYDRO_frocing_drv (indir,forc_typ, snow_assim,olddate,     &
           T2,Q2X,U,V,PRES,XLONG,SHORT,PRCP1,lai,fpar,snodep, kt, FORCING_TIMESTEP,pcp_old)
 
          use module_lsm_forcing, only: read_hydro_forcing
-         use module_namelist, only:nlst_rt
+         !use module_namelist, only:nlst_rt
```
Comment (jmccreight) [inline trunk/NDHMS/CPL/NoahMP_cpl/hrldas_drv_HYDRO.F]: remove

File: trunk/NDHMS/CPL/NoahMP_cpl/module_hrldas_HYDRO.F [hunk 4]
```diff
@@ -29,7 +29,8 @@ module module_HRLDAS_HYDRO
 #endif
     use module_HYDRO_drv, only: HYDRO_ini, HYDRO_exe, HYDRO_rst_out
     use module_rt_data, only:  rt_domain
-    use module_namelist, only: nlst_rt
+    !use module_namelist, only: nlst_rt
```
Comment (jmccreight) [inline trunk/NDHMS/CPL/NoahMP_cpl/module_hrldas_HYDRO.F]: remove

File: trunk/NDHMS/CPL/Noah_cpl/hrldas_drv_HYDRO.F [hunk 5]
```diff
@@ -52,7 +52,9 @@ subroutine HYDRO_forcing_drv (indir,forc_typ, snow_assim,olddate,     &
           T2,Q2X,U,V,PRES,XLONG,SHORT,PRCP1,lai,fpar,snodep, kt,FORCING_TIMESTEP,prcp_old)
 
          use module_lsm_forcing, only: read_hydro_forcing, read_hydro_forcing_seq
-         use module_namelist, only:nlst_rt
+         !use module_namelist, only:nlst_rt
```
Comment (jmccreight) [inline trunk/NDHMS/CPL/Noah_cpl/hrldas_drv_HYDRO.F]: remove

File: trunk/NDHMS/OrchestratorLayer/config.f90 [hunk 6]
```diff
@@ -0,0 +1,924 @@
...
+   contains
+     procedure, nopass :: init => config_init
+     procedure, nopass :: init_nlst => init_namelist_rt_field
+  end type Configuration_
+
+  integer, parameter :: max_domain = 5
+
+  type(NOAHLSM_OFFLINE_), protected, save :: noah_lsm
+  type(WRF_HYDRO_OFFLINE_), protected, save :: wrf_hydro
+  type(namelist_rt_), dimension(max_domain), save :: nlst
```
Comment (rcabell) [inline trunk/NDHMS/OrchestratorLayer/config.f90]: Should this be `protected`? Why not? Ideally, this should be read-only.

Comment (afanfa) [reply trunk/NDHMS/OrchestratorLayer/config.f90]: Yes, I agree that it should be read-only. At the moment, DTCT and sincedate are read from file using netcdf in module_NWM_io.F. DTCT is also manipulated with conditions like the following:
nlst(did)%DTCT = min(nlst(did)%DTCT, nlst(did)%DTRT_CH).
What do you recommend? Creating a private copy of the namelist, moving these changes into the config.f90?

**Comment by rcabell** [comment]
As expected, this PR passes all CONUS Integration tests. 

Interesting results in timing, however:

* for a 48-hour AnA run with once-daily restarts, this code runs about 20% slower
* for a 4-day run with hourly restarts, this code runs about 25% _faster_

Strange, eh?

Recommend closing.

**Comment by afanfa** [comment]
@rcabell could it be that the performance difference is just related to Cheyenne's instability?

**Comment by rcabell** [comment]
> @rcabell could it be that the performance difference is just related to Cheyenne's instability?

Yeah, it very well could be. I actually see a pretty consistent 15-20% improvement when testing locally...

**Pull request merged**